WalletWasabi.Gui/Controls/WalletExplorer/SendTabViewModel.cs

```diff
@@ -52,7 +52,24 @@ protected override async Task BuildTransaction(string password, PaymentIntent pa
 					PSBT signedPsbt = null;
 					try
 					{
-						signedPsbt = await client.SignTxAsync(Wallet.KeyManager.MasterFingerprint.Value, result.Psbt, cts.Token);
+						try
+						{
+							signedPsbt = await client.SignTxAsync(Wallet.KeyManager.MasterFingerprint.Value, result.Psbt, cts.Token);
+						}
+						catch (PSBTException ex) when (ex.Message.Contains("NullFail"))
+						{
+							NotificationHelpers.Warning("Fall back to Unverified Inputs Mode, trying to sign again.");
+
+							// Ledger Nano S hackfix https://github.com/MetacoSA/NBitcoin/pull/888
+
+							var noinputtx = result.Psbt.Clone();
```

Comment:
Rename `noinputtx` to `noInputPsbt`.

BTW why clone the psbt? what if there is an exception and line 77 is reached?

Reply:
> BTW why clone the psbt? what if there is an exception and line 77 is reached?

I suppose, "just to be sure".